linked_list/node.py

```diff
@@ -4,13 +4,19 @@ class Node(object):
     """
 
     def __init__(self, elem, next=None):
-        pass
+        self.elem = elem
+        self.next = next
 
     def __str__(self):
-        pass
+        return str(self.elem)
 
     def __eq__(self, other):
-        pass
+        
+        #if self.elem is not None and other.elem is not None:
+        if type(other) is Node:
+            return self.elem == other.elem
+        else:
+            return self.elem == other
```

Comment:
Why are you comparing element with other if their type doesn't match?
Shouldn't we return `False` for `Node(3) == 3`?